vector-im/element-android
# #5395: Location sharing options view

## Type of change


## Content


Adding a new custom view named `LocationSharingOptionPickerView` to be able to show different options of location sharing. 
The only visible changes we can see now in the app is the icon and text displayed to confirm sharing and the avatar display in timeline and map (see captures below). 
In fact, we only have user current location option available for now. Icon is now the user avatar with background color corresponding to the user color both in the pins of the map and in the confirmation view. Ripple effect has also been added when the confirmation view is pressed.

## Motivation and context


Closes #5395.

## Screenshots / GIFs



### Current visible changes in the app

<img src="https://user-images.githubusercontent.com/46314705/156542888-b3894879-22f5-4346-88b3-57874d3c06a4.png" width=300/><img src="https://user-images.githubusercontent.com/46314705/156542912-100eb2d4-b478-4b1d-ac57-5e72d3fa3ac9.png" width=300/>

<img src="https://user-images.githubusercontent.com/46314705/156545861-e87c3782-5cc7-45d9-be94-c312e69f0cac.png" width=300/><img src="https://user-images.githubusercontent.com/46314705/156545880-7da06fe6-36db-4fde-8983-4f1af6a3bcfd.png" width=300/>

### Changes not currently visible in the app

For UI validation, **only the option picker view is relevant, the map view has not been updated for now)**

<img src="https://user-images.githubusercontent.com/46314705/156546246-b5da868c-95d0-4efe-9cb3-27a654429e10.png" width=300/><img src="https://user-images.githubusercontent.com/46314705/156546285-23e3e87e-4797-4a53-ab8b-b7e537cb9bf3.png" width=300/>

<img src="https://user-images.githubusercontent.com/46314705/156546827-f17f5ee7-d21b-4042-aaf5-186eec158733.png" width=300/><img src="https://user-images.githubusercontent.com/46314705/156546853-afee92a4-d028-40cc-b6b6-136628c2cbb2.png" width=300/>

## Tests



- Ensure location sharing option is enabled in "Settings" -> "Preferences"
- Go to a room or a direct message
- Click on the "add" icon
- Select the location icon
- Check you see the map with your current location
- Check you see the confirmation view with your avatar and color
- Check the avatar background color is yours and consistent both in the map and in the confirmation view
- Click on the confirmation view
- Check the location is correctly shared in the timeline of the room/message
- Check the avatar background color is yours and consistent in the timeline

To test the other options, go to `LocationSharingFragment` at `initOptionsPicker()` method and set the options as you want. For example, `views.shareLocationOptionsPicker.render(LocationSharingOption.USER_CURRENT, LocationSharingOption.USER_LIVE)` or  `views.shareLocationOptionsPicker.render(LocationSharingOption.PINNED)`

## Tested devices

- OS version(s): Android 11

## Checklist
